### ISSUE TITLE
Is cmake 3.9 really a requirement?

### ISSUE THREAD

**Issue by ekg** [alancleary/libvgio#5]
```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.9 or higher is required.  You are running version 3.3.0-rc2
```

So I should recompile cmake? What features do you need from the diff between 3.3 and 3.9?

**Comment by alancleary** [comment]
I just tried to use cmake version 3.3.0-r2. It's not failing on a more recent feature, per se, but it chokes on .c files. Apparently the CXX language as defined in 3.3.0-r2 doesn't allow .c files to be compiled as C++. This can be remedied by [adding the files in question to the CXX language](https://stackoverflow.com/a/7693134). That said, I think it would be prudent to set the minimum cmake version to a version where this isn't an issue.

Perhaps @adamnovak has an opinion on the matter.

**Comment by adamnovak** [comment]
I wanted to use 3.13 for `target_link_directories`, but that's not out yet in a Cmake in Ubuntu, so I worked around it. I think I fell back to 3.9 because that was the version Travis uses when building vg, so it's the oldest Cmake actually under CI. I wanted to avoid a requirement lower than what was actually required to build successfully.

We can probably lower it further, if we can test it adequately. I believe @alancleary originally had 3.6 as the minimum version before I changed the CMakeLists.txt all around.

**Comment by alancleary** [comment]
Yes, I did have it set to 3.6 (for no particular reason), though I was actually building with 3.14. Apparently support for specific language standard levels (e.g. `cxx_std_14`) wasn't added to Compile Features [until version 3.8](https://cmake.org/cmake/help/latest/release/3.8.html#c-c), so building with 3.6 would've actually thrown an error... If we want to specify a minimum version lower than 3.8, we would have to set the language standard [with some `CMAKE_CXX_*` variables](https://stackoverflow.com/a/42842447).

Regarding Travis, I think as long as the minimum cmake version specified is less than what Travis is using then we should be good to go. It would probably be best to set the version as low as we deem reasonable so we can support as many systems as possible. For instance, Ubuntu 16.04 [still has a couple more years of maintenance updates](https://www.ubuntu.com/about/release-cycle), so it's quite possible that people are still using it. Its [cmake package](https://packages.ubuntu.com/xenial/cmake) is currently at version 3.5.1. I've confirmed that by using the `CMAKE_CXX_*` variables the project builds just fine with cmake version 3.5.1. So yeah, food for thought.
